includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1342,8 +1341,11 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 			} else {
 				$save_user_in_platform_checkout = false;
 
+				$session_data = WC()->session->get( 'platform-checkout-user-data' );
 				// phpcs:ignore WordPress.Security.NonceVerification.Missing,WordPress.Security.ValidatedSanitizedInput.MissingUnslash
-				if ( ! empty( $_POST['save_user_in_platform_checkout'] ) && filter_var( $_POST['save_user_in_platform_checkout'], FILTER_VALIDATE_BOOLEAN ) ) {
+				$should_save_platform_customer = ( ! empty( $_POST['save_user_in_platform_checkout'] ) && filter_var( $_POST['save_user_in_platform_checkout'], FILTER_VALIDATE_BOOLEAN ) ) || ( isset( $session_data['save_user_in_platform_checkout'] ) && filter_var( $session_data['save_user_in_platform_checkout'], FILTER_VALIDATE_BOOLEAN ) );
+
+				if ( $should_save_platform_customer ) {
```

Comment:
```suggestion
				if ( $payment_information->should_save_payment_method_to_platform() ) {
```
Can't we just rely on the value that you set above on the `Payment_Information` instance now? This looks like exactly the same check that you perform over [here](https://github.com/Automattic/woocommerce-payments/blob/task/4614-new-platform-checkout-user-from-checkout-blocks/includes/class-wc-payment-gateway-wcpay.php#L1045), to my untrained eyes.

Reply:
Your untrained eyes are very powerful :open_mouth: 
I have updated this to use the new utils method you suggested in [another comment](https://github.com/Automattic/woocommerce-payments/pull/4638#discussion_r988538727).